src/main/java/frc/robot/subsystems/CoralJoint.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.spark.SparkLowLevel;
+import com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem.SmartMotorControllerGenericSubsystem;
+import com.spikes2212.util.smartmotorcontrollers.SparkWrapper;
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.robot.RobotMap;
+
+public class CoralJoint extends SmartMotorControllerGenericSubsystem {
```

Comment:
add distance per pulse (which should be the gear ratio multiplied by 360 (save that as a constant too))